### ISSUE TITLE
Compatibility with Nextcloud 25

### ISSUE THREAD

**Issue by Supergamerx3000** [SergeyMosin/Appointments#366]
Nextcloud 25 RC2 has been released. It will be released at the end of October. Please create a compatible release for Nextcloud 25.

Kind regards
Pascal

**Comment by vithusel** [comment]
Is this plugin any closer to reaching v25 compatibility? 

V25 went live today

**Comment by RoundAboutWEB** [comment]
IMPORTANT: When can compatibility be expected?

**Comment by SergeyMosin** [comment]
Testing is planned over the next couple of days and if everything goes well the next version will be release on Oct 24

**Comment by support-tt** [comment]
@RoundAboutWEB dont hurry him he is developing this app for free in his free time so sometimes a update needs some time..

@SergeyMosin I installed the app in our test enviroment the fields are all top left of the site and not well formatted but the app itself seems to work. Maybe its only a small change to support 25

**Comment by vithusel** [comment]
> Testing is planned over the next couple of days and if everything goes well the next version will be release on Oct 24

Awesome! Thanks for the update. 👍

**Comment by SergeyMosin** [comment]
Just a quick update:
It looks like there are major changes on the front-end in NC25 https://github.com/nextcloud/server/issues/32117 and it might break custom CSS that users might have added to their appointment pages. Therefore update is going to be two-fold:
1. I'll push the first update today with just a message/notice informing users about possible custom CSS breakage in the next update.
2.  On Friday I'll push the real update

**Comment by RoundAboutWEB** [comment]
**So we hope for 30.10.22:**

![image](https://user-images.githubusercontent.com/26809494/198171622-55568300-faeb-4392-af5c-5172b2db2454.png)

**Comment by SergeyMosin** [comment]
v1.14.0 is NC25 compatible.

App UI still uses `"@nextcloud/vue": "^5.4.0"` so no fancy transparency effects for now. Style for the public page is kept as close to the original as possible, but if you want to use the new NC "frost/blur" effects you can use the following CSS Overrides:
```
<style>
body {
  background-image: var(--image-main-background);
}
#header {
  background: none;
}
#srgdev-ncfp_frm {
  background-color: var(--color-main-background-blur);
  -webkit-backdrop-filter: var(--filter-background-blur);
  backdrop-filter: var(--filter-background-blur);
  border-radius: 10px;
  padding: 2em;
}
@media only screen and (max-width: 390px) {
  #srgdev-ncfp_frm {
    padding: 1em;
    margin-left: 0;
    margin-right: 0;
  }
  .srgdev-ncfp-wrap {
    font-size: 105%;
  }
}
</style>
```
![image](https://user-images.githubusercontent.com/10747559/198862531-e57a73cf-5fc6-41f1-aa12-06d125a68be4.png)

**Comment by RoundAboutWEB** [comment]
In witch file?

**Comment by SergeyMosin** [comment]
@RoundAboutWEB CSS ? `Settings > Customize Public Page > Advanced Settings > Style Override`

**Comment by SergeyMosin** [comment]
In order to comply with accessibly guidelines keyboard navigation is implemented in 1.14.6 ref: #214

Additionally, for performance and stability in `Auto Style` functionality and to avoid crashes in some edge cases `v1.14.2` and `v1.14.3` users should update the app to `v1.14.4+` or turn off `Auto Style` until then

**Comment by SergeyMosin** [comment]
v1.14.7 has further improvements to 'Auto Style` module. No other NC25 related issues are found. Closing.
